core/extension/gdextension_interface.cpp

```diff
@@ -1572,6 +1572,33 @@ static GDExtensionScriptInstancePtr gdextension_object_get_script_instance(GDExt
 	return script_instance_extension->instance;
 }
 
+static void gdextension_object_set_script_instance(GDExtensionObjectPtr p_object, GDExtensionConstObjectPtr p_language, GDExtensionScriptInstancePtr p_script_instance) {
+	ERR_FAIL_NULL(p_object);
+	ERR_FAIL_NULL(p_language);
+
+	Object *o = (Object *)p_object;
+	ScriptLanguage *language = (ScriptLanguage *)p_language;
+	ScriptInstance *script_instance = (ScriptInstanceExtension *)p_script_instance;
+
+	Ref<Script> script = o->get_script();
+	ERR_FAIL_COND(script.is_null());
+	ERR_FAIL_COND(script->get_language() != language);
+
+	o->set_script_instance(script_instance);
+}
+
+static void gdextension_object_set_script_and_instance(GDExtensionObjectPtr p_object, GDExtensionObjectPtr p_script, GDExtensionScriptInstancePtr p_script_instance) {
```

Comment:
Is there a reason we pass `p_script` instead of relying on `script_instance->get_script()` ?

According to `Object::set_script_and_instance`,  `p_script` and `p_script_instance` are not allowed to be `NULL`:

https://github.com/godotengine/godot/blob/eee39f004b42a4948af90cdebedf65c34a4a4970/core/object/object.cpp#L951-L955

So having `p_script` parameter only allows us to have `p_script != p_script_instance->get_script()` which I fail to see any benefit of having :thinking:

Reply:
Yeah, based on the discussion above, I don't think we need `object_set_script_and_instance`